viadot/sources/databricks.py

```diff
@@ -44,146 +57,160 @@ def _get_spark_session(self, env: str = None):
-        session = SparkSession(context)
-        new_spark = session.builder.config(conf=new_config).getOrCreate()
-        return new_spark
-
     def connect(self):
         if self.env == "QA":
-            self.spark = SparkSession.builder.getOrCreate()
+            self.sc = SparkSession.builder.getOrCreate()
         else:
-            self.spark = self._get_spark_session(self.env)
-    
-    def to_df(self, list):
-        return pd.DataFrame(list)
-    
-    def to_json(sparkDf: pyspark.sql.dataframe.DataFrame):
-        return sparkDf.toJSON()
-    
-    def to_spark_df(self, df: DataFrame):
+            self.sc = self._get_spark_session(self.env)
+
+    def to_df(self, table_name: str, if_empty="fail"):
         """
-        Convert Pandas DataFrame to Spark DataFrame
+        Read the contents of a given table and return it as a Pandas DataFrame
 
         Args:
-            df (DataFrame): Pandas DataFrame to be converted to Spark DataFrame.
+            table_name (str): Name of an existing table.
 
         Example:
         ```python
         from viadot.sources import Databricks
         databricks = Databricks()
-        list = [{"some":"data"}]
-        sparkDf = databricks.to_spark_df(list)
+        table_data = databricks.to_df("schema.table_1")
         ```
         Returns:
-            pyspark.sql.dataframe.DataFrame: The converted Spark DataFrame
+            pd.DataFrame: A Pandas DataFrame containing the requested table's data.
         """
-        sparkDF = self.spark.createDataFrame(df)
-        return sparkDF
-    
-    def spark_to_pandas(self, sparkDf: pyspark.sql.dataframe.DataFrame):
+        result = self.sc.sql("select * from " + table_name)
+        df = self._spark_df_to_pandas_df(result)
+        if df.empty:
+            self._handle_if_empty(if_empty)
+        return df
+
+    def to_json(self, table_name: str, if_empty="fail"):
         """
-        Convert Spark DataFrame to Pandas DataFrame
+        Read the contents of a given table and return it as a JSON string
 
         Args:
-            df (pyspark.sql.dataframe.DataFrame): Spark DataFrame to be converted to Pandas DataFrame.
+            table_name (str): Name of an existing table.
 
         Example:
         ```python
         from viadot.sources import Databricks
         databricks = Databricks()
-        list = [{"some":"data"}]
-        sparkDf = databricks.to_spark_df(df)
-        pandasDf = spark_to_pandas(sparkDf)
+        table_data = databricks.to_json("schema.table_1")
         ```
         Returns:
-            pyspark.sql.dataframe.DataFrame: The converted Spark DataFrame
+            str: A JSON string containing the requested table's data.
         """
-        return sparkDf.toPandas()
-    
-    def query(self, query: str):
+        result = self.sc.sql("select * from " + table_name)
+        df = self._spark_df_to_pandas_df(result)
+        if df.empty:
+            self._handle_if_empty(if_empty)
+        return df.to_json()
```

Comment:
The purpose of `to_json()` is to offer a more flexible way to consume data (not necessarily a pandas DataFrame). If you need to first load it to a DataFrame anyway then this function doesn't make sense -- user can just use `Databricks.to_df()`.